go/cmd/dolt/commands/migrate.go

```diff
@@ -45,31 +55,218 @@ func (cmd MigrateCmd) CreateMarkdown(_ filesys.Filesys, _, _ string) error {
+	switch {
+	case apr.Contains(migratePushFlag):
+		err = pushMigratedRepo(ctx, dEnv, apr)
+	case apr.Contains(migratePullFlag):
+		err = fetchMigratedRemoteBranches(ctx, dEnv, apr)
+	default:
+		err = migrateLocalRepo(ctx, dEnv)
 	}
 
-	cli.Println(color.YellowString(migrationMsg))
-	err = rebase.MigrateUniqueTags(ctx, dEnv)
 	if err != nil {
-		cli.PrintErrf("error migrating repository: %s", err.Error())
+		cli.PrintErrln(color.RedString(err.Error()))
 		return 1
 	}
+
 	return 0
 }
 
+func migrateLocalRepo(ctx context.Context, dEnv *env.DoltEnv) error {
+	localMigrationNeeded, err := rebase.NeedsUniqueTagMigration(ctx, dEnv.DoltDB)
+
+	if err != nil {
+		return err
+	}
+
+	if localMigrationNeeded {
+		cli.Println(color.YellowString(migrationMsg))
+		err = rebase.MigrateUniqueTags(ctx, dEnv)
+
+		if err != nil {
+			return err
+		}
+	} else {
+		cli.Println("Repository format is up to date")
+	}
+
+	remoteName := "origin"
+	remoteMigrated, err := remoteHasBeenMigrated(ctx, dEnv, remoteName)
+	if err != nil {
+		// if we can't check the remote, exit silently
+		return nil
+	}
+
+	if !remoteMigrated {
+		cli.Println(fmt.Sprintf("Remote %s has not been migrated", remoteName))
+		cli.Println(fmt.Sprintf("Run 'dolt migrate --push' %s to update remote", remoteName))
+	} else {
+		cli.Println(fmt.Sprintf("Remote %s has been migrated", remoteName))
+		cli.Println(fmt.Sprintf("Run 'dolt migrate --pull %s' to update refs", remoteName))
+	}
+
+	return nil
+}
+
+func pushMigratedRepo(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults) error {
+	localMigrationNeeded, err := rebase.NeedsUniqueTagMigration(ctx, dEnv.DoltDB)
+	if err != nil {
+		return err
+	}
+	if localMigrationNeeded {
+		return fmt.Errorf("Local repo must be migrated before pushing, run 'dolt migrate'")
+	}
+
+	remoteName := "origin"
+	if apr.NArg() > 0 {
+		remoteName = apr.Arg(0)
+	}
+
+	remotes, err := dEnv.GetRemotes()
+	if err != nil {
+		return err
+	}
+
+	remote, remoteOK := remotes[remoteName]
+	if !remoteOK {
+		return fmt.Errorf("unknown remote %s", remoteName)
+	}
+
+	remoteMigrated, err := remoteHasBeenMigrated(ctx, dEnv, remoteName)
+	if err != nil {
+		return err
+	}
+	if remoteMigrated {
+		return fmt.Errorf("Remote %s has been migrated\nRun 'dolt migrate --pull' to update refs", remoteName)
+	} else {
+		// force push all branches
+		bb, err := dEnv.DoltDB.GetBranches(ctx)
+
+		if err != nil {
+			return err
+		}
+
+		for _, branch := range bb {
+			refSpec, err := ref.ParseRefSpec(branch.String())
+			if err != nil {
+				return err
+			}
+
+			src := refSpec.SrcRef(branch)
+			dest := refSpec.DestRef(src)
+
+			remoteRef, err := getTrackingRef(dest, remote)
+
+			if err != nil {
+				return err
+			}
+
+			destDB, err := remote.GetRemoteDB(ctx, dEnv.DoltDB.ValueReadWriter().Format())
+
+			if err != nil {
+				return err
+			}
+
+			cli.Println(color.YellowString(fmt.Sprintf("Pushing migrated branch %s to %s", branch.String(), remoteName)))
```

Comment:
Is this a warning message or an information message?  If it's an information message don't think I would color it yellow.